actions/treasures.go

```diff
@@ -15,7 +24,51 @@ type treasureReq struct {
 	EthKey          string `json:"EthKey"`
 }
 
+type treasureRes struct {
+	Success string `json:"success"`
+}
+
+func unixMilli(t time.Time) int64 {
+	return t.Round(time.Millisecond).UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+}
+
 // Verifies the treasure and claims such treasure.
 func (t *TreasuresResource) VerifyAndClaim(c buffalo.Context) error {
-	return nil
-}
+	req := treasureReq{}
+	oyster_utils.ParseReqBody(c.Request(), &req)
+
+	host_ip := os.Getenv("HOST_IP")
+	provider := "http://" + host_ip + ":14265"
+	api := giota.NewAPI(provider, nil)
+	ftr := &giota.FindTransactionsRequest{Bundles: []giota.Trytes{"DEXRPLKGBROUQMKCLMRPG9HFKCACDZ9AB9HOJQWERTYWERJNOYLW9PKLOGDUPC9DLGSUH9UHSKJOASJRU"}}
```

Comment:
I don't think you will be looking for Bundles.  You'll be looking for addresses.

Reply:
Is FindTransactionsRequest correct? How it should looks? Should it be like Address: [].giota.toAddress{req.GenesisHash}  ?

How can I transform result from FindTransactionsRequest to datamap or array (then I would like to walk through)

Reply:
FindTransactionsRequest is correct for searching transactionsObjects? 
Commad: "findTransactions" or Command:   "findTransactionsObjects”?

If I dont want to search by bundles how it should looks?

Is this example correct for validation with timestamp?

chunks []models.DataMap
addresses := make([]giota.Address, 0, len(chunks))
for _, chunk := range chunks {
	if chunk.Status != models.Error {
		addresses = append(addresses, giota.Address(chunk.Address))
	}
}
request := giota.FindTransactionsRequest{
	Command:   "findTransactions", 
	Addresses: addresses,
}
response, err := api.FindTransactions(&request)

trytesArray, err := api.GetTrytes(response.Hashes)

transactionObjects := map[giota.Address][]giota.Transaction{}

for _, txObject := range trytesArray.Trytes {
	 transactionObjects[txObject.Address].timestamp  — timestamp???
}

Reply:
Here's how to build the FindTransactionRequest:
https://godoc.org/github.com/iotaledger/iota.lib.go#FindTransactionsRequest   

As for the address, you'll get those when you build the dataMap.  

It'll be something like this:  
```
sectorDataMap := BuildSectorDataMap(genesisHash, sector, numChunks)
   /* BuildSectorDataMap will be some new method you write that does not save to the data_maps table
Only need to build the dataMap up until the end of the sector the webnode did.  Need numChunks to know when to stop building the dataMap, in the event that the sector the webnode passed was the last sector.  I.e. if there are only 2,500,000 hashes in the dataMap and the webnode did Sector 3, need the numChunks so we know to stop at 2,500,000.   

BuildSectorDataMap would not need to do the obfuscation and tryting for the whole map, just the section that you care about.  You'll still have to get the sha256 of all the chunks leading up to the sector you care about.  So let's say it's sector 3 in a file with 2,500,000 chunks.  You'll have to get the sha256 hashes of chunks 0 through 1,999,999, but you'll only have to do the obfuscation and trying for hashes 2,000,000 through 2,500,000.  The method could just return the values for 2,000,000 thru 2,500,000, don't need ALL the addresses for the other sectors.  
*/


addresses := make([]giota.Address, 0, len(sectorDataMap))
for _, sectorDataMapAddress := range sectorDataMap {
addresses = append(addresses, giota.Address(sectorDataMapAddress))
}
request := giota.FindTransactionsRequest{
Addresses: addresses,
}
response, err := api.FindTransactions(&request)

trytesArray, err := api.GetTrytes(response.Hashes)

/* You're on the right track here.  You basically need to verify this:   
1.  For every address in addresses, there is a transaction in trytesArray.Trytes, and there is at least one transaction with that address from within the last epoch (there could be more than 1 with the same address)
*/

//transactionObjects := map[giota.Address][]giota.Transaction{}
transactionInLastEpoch := map[giota.Address][]bool
^You will have to experiment with this

// You could do something like this:  
for _, address := range addresses {
       transactionInLastEpoch[address] = false
}

for _, txObject := range trytesArray.Trytes {
   if txObject.timestamp > timeWeAreComparingAgainst {
      transactionInLastEpoch[txObject.Address] = true
   }
}

/* then iteration through transactionInLastEpoch.  If ALL values are true, the webnode did the PoW.  If any values are false, it lied.  */

```